lib/app.js

```diff
@@ -294,13 +294,20 @@ function uploadSite(settings, directoryRef, argv) {
 }
 
 function getSettings(argv) {
-  var settingsFile = path.resolve('./firebase.json');
-  var settingsJSON, settings;
-  if (!fs.existsSync(settingsFile)) {
-    console.log(chalk.red('Initialization Error') + ' - Directory not ' +
+  var curDir = process.cwd();
+  while (!fs.existsSync(path.resolve(curDir, './firebase.json'))) {
```

Comment:
This is more readable now and acceptable, thank you. As an aside - it's unlikely that this would have been called more than 10 times deep ever, so the lack of tail recursion in JavaScript would have had negligible effect